test/lit/passes/global-effects-O.wast

```diff
@@ -0,0 +1,204 @@
+;; NOTE: Assertions have been generated by update_lit_checks.py --all-items and should not be edited.
+
+;; RUN: foreach %s %t wasm-opt -all -S -o - --generate-global-effects     | filecheck %s --check-prefix CHECK_0
+;; RUN: foreach %s %t wasm-opt -all -S -o - --generate-global-effects -O1 | filecheck %s --check-prefix CHECK_1
+;; RUN: foreach %s %t wasm-opt -all -S -o - --generate-global-effects -O3 | filecheck %s --check-prefix CHECK_3
+;; RUN: foreach %s %t wasm-opt -all -S -o - --generate-global-effects -Os | filecheck %s --check-prefix CHECK_s
+;; RUN: foreach %s %t wasm-opt -all -S -o - --generate-global-effects -O  | filecheck %s --check-prefix CHECK_O
+
+;; Test that global effects benefit -O1 and related modes.
+
+(module
+  ;; CHECK_0:      (type $0 (func))
+
+  ;; CHECK_0:      (type $1 (func (result i32)))
+
+  ;; CHECK_0:      (export "export" (func $export))
+
+  ;; CHECK_0:      (export "pointless-work" (func $pointless-work))
```

Comment:
If we put an explicit `(export "pointless-work" (func $pointless-work))` here, then that and everything above it in the output would be isolated here and only the various `(func $export ...)` implementation would be down above that function.

Reply:
Nice, thanks, that makes the output much cleaner.